### PR TITLE
Sample: custom riscv32 bitcode for the mmt4d ukernel

### PR DESCRIPTION
**Pull request by bjacob**
This shows how to develop out-of-tree custom bitcode for the standard mmt4d ukernel, here for the riscv32 architecture.

I haven't been able to test locally, so there might be a bug. At least, it builds, and the disassembly seems sane.

@lundong @hcindyl 

I'll be OOO until August 20 . In the interim @benvanik and @MaheshRavishankar can answer questions :-)

### CODE REVIEW

File: samples/ukernel_mmt4d_riscv_32/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+iree_add_all_subdirs()
```
Comment (benvanik) [inline samples/ukernel_mmt4d_riscv_32/CMakeLists.txt:1]: not sure if you plan to land this or just treat it as an example PR, but if landing you'll want to predicate all this on whether the cpu backend is enabled and probably some other things - something like

```
iree_compiler_targeting_iree_arch(_IREE_COMPILER_TARGETING_RISCV_32 "riscv_32")
if(NOT IREE_TARGET_BACKEND_LLVM_CPU OR
   NOT _IREE_COMPILER_TARGETING_RISCV_32 OR
   NOT IREE_HAL_DRIVER_LOCAL_TASK OR
   NOT IREE_HAL_EXECUTABLE_LOADER_EMBEDDED_ELF)
  return()
endif()
```

Comment (benvanik) [inline samples/ukernel_mmt4d_riscv_32/CMakeLists.txt:1]: let's throw this under `samples/custom_dispatch/cpu/ukernel/mmt4d/` so we have one place for these (I'd like to rework the other samples to show ukernels mixed in as well, and it'll give us a place to put other examples as well)

**Review by benvanik — CHANGES_REQUESTED** [review]
nice!

File: samples/ukernel_mmt4d_riscv_32/CMakeLists.txt (hunk 1 above)
Comment (bjacob) [reply samples/ukernel_mmt4d_riscv_32/CMakeLists.txt:1]: yeah this was not meant for landing -- stripped this kind of boilerplate to make this minimal.  agree we readd it if landing.

**Pull request closed without merging**